packages/core/utils/src/payment/abstract-payment-provider.ts

```diff
@@ -678,6 +680,52 @@ export abstract class AbstractPaymentProvider<TConfig = Record<string, unknown>>
     context: PaymentProviderContext
   ): Promise<PaymentMethodResponse[]>
 
+  /**
+   * Saves a payment method associated with the context (eg. customer) of the payment provider.
+   *
+   * @param input -The data for the request and context for which the payment method is saved. Usually the customer should be provided.
+   * @returns An object whose `data` property is set to the data returned by the payment provider.
+   *
+   * @example
+   * // other imports...
+   * import {
+   *   SavePaymentMethod,
+   *   PaymentProviderError,
+   *   SavePaymentMethodResponse
+   * } from "@medusajs/framework/types"
+   *
+   *
+   * class MyPaymentProviderService extends AbstractPaymentProvider<
+   *   Options
+   * > {
+   *   async savePaymentMethod(
+   *     input: SavePaymentMethod
+   *   ): Promise<PaymentProviderError | SavePaymentMethodResponse> {
+   *     const {context, data } = input
+   *     const externalCustomerId = context.customer.metadata.stripe_id
+   *
+   *     if(!externalCustomerId) {
+   *       return {
+   *         error: new Error("Customer ID is required"),
+   *         code: "unknown",
+   *         detail: "Customer ID is required"
+   *       }
+   *     }
+   *
+   *     const response = await this.client.savePaymentMethod(
+   *       {customer: externalCustomerId, ...data}
+   *     )
+   *
+   *     return response
+   *   }
+   *
+   *   // ...
+   * }
+   */
+  abstract savePaymentMethod(
```

Comment:
Q: should we make this optional? I believe not all provider can implement that right?

Reply:
Yeah sounds good, I made `listPaymentMethods` optional as well as not all providers would support it.